src/SkyApm.Diagnostics.MassTransit/Observers/MasstransitReceiveObserver.cs

```diff
@@ -0,0 +1,100 @@
+﻿using MassTransit;
+using SkyApm.Common;
+using SkyApm.Config;
+using SkyApm.Diagnostics.MassTransit.Common;
+using SkyApm.Tracing;
+using SkyApm.Tracing.Segments;
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace SkyApm.Diagnostics.MassTransit.Observers
+{
+    public class MasstransitReceiveObserver : IReceiveObserver
+    {
+        private readonly ITracingContext _tracingContext;
+        private IEntrySegmentContextAccessor _entrySegmentContextAccessor;
+        private TracingConfig _tracingConfig;
+        public MasstransitReceiveObserver(ITracingContext tracingContext,
+            IEntrySegmentContextAccessor entrySegmentContextAccessor,
+            IConfigAccessor configAccessor)
+        {
+            _tracingContext = tracingContext;
+            _entrySegmentContextAccessor = entrySegmentContextAccessor;
+            _tracingConfig = configAccessor.Get<TracingConfig>();
+        }
+
+        public Task PreReceive(ReceiveContext context)
+        {
+            //activity will be null in aspnet.core web
+            //we need to config the ActivityListener
+            ///<seealso cref="MasstransitSkyApmHostingStartup"/>
+            var activity = Activity.Current ?? default;
+
+            var segContext = _tracingContext.CreateEntrySegmentContext("Masstransit Receiving/ " + activity.OperationName,
+                new MasstransitCarrierHeaderCollection(context.TransportHeaders));
+            segContext.Span.SpanLayer = SpanLayer.DB;
```

Comment:
The DB call cannot be an entry span, and an exit span should be used

Reply:
Any more pre-cautions to use the Span.SpanLayer?  Just fount how to use Span.Component. Thanks.